server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -162,6 +169,18 @@ protected SearchHit nextDoc(int doc) throws IOException {
                 if (context.isCancelled()) {
                     throw new TaskCancelledException("cancelled");
                 }
+                ++processedDocs;
+                // indicates if we should check with the parent circuit breaker (the real memory breaker) if we have enough memory to
+                // continue the fetch phase.
+                // we do so whenever one of the following is true:
+                // 1. we have accumulated at least the size of the memory accounting buffer
+                // 2. we have reached the last document in the leaf
+                boolean enoughBytesOrLastDocInLeaf = (accumulatedBytesInLeaf >= context.memAccountingBufferSize())
```

Comment:
NIT: two redundant sets of brackets here :) also maybe just inline the `boolean`, with all the text documentation the variable name real doesn't add anything ?